compiler/internal/codegen/codegen_main.go

```diff
@@ -142,7 +142,10 @@ func (b *Builder) Main() (f *File, err error) {
 	f.Line()
 
 	f.Func().Id("main").Params().Block(
-		If(Err().Op(":=").Qual("encore.dev/runtime", "ListenAndServe").Call(), Err().Op("!=").Nil()).Block(
+		If(
+			Err().Op(":=").Qual("encore.dev/runtime", "ListenAndServe").Call(),
+			Err().Op("!=").Nil().Op("&&").Err().Op("!=").Qual("io", "EOF"),
```

Comment:
Ths addition of the `&& err != io.EOF` stops the Encore app from reporting a fatal error when it's shutdown cleanly